src/services/category.js

```
@@ -25,27 +26,49 @@ const categoryService = {
 
-    const sortOptions = sort?.orderBy ? { [sort.orderBy]: sort.order?.toLowerCase() === 'desc' ? -1 : 1 } : {}
+    const [data, total] = await Promise.all([
+      Category.find(query).sort(sort).skip(skip).limit(limit).lean(),
+      Category.countDocuments(query)
+    ])
 
-    return await Category.find(query).sort(sortOptions).skip(Number(skip)).limit(Number(limit))
+    const mappedData = data.map(({ _id, name, appearance }) => ({ id: _id.toString(), name, appearance }))
+
+    return {
+      data: mappedData,
```

ESLint

[error] Expected linebreaks to be 'CRLF' but found 'LF'.

(linebreak-style)